### ISSUE TITLE
Enhancement - Add CCI30 as possible benchmark / Add "Buy and Hold" comparison

### ISSUE THREAD

**Issue by Dante2333** [freqtrade/freqtrade#1913]
Add CCI30 (cci30.com) as possible benchmark comparison for strategies on backtesting period
Add Buy and Hold comparison of given currencies to compare to the strategy

**Comment by hroff-1902** [comment]
CCI30 is a composite index. How do you calculate it for a strategy if I trade only ETH/BTC on a given exchange, for example?

**Comment by Dante2333** [comment]
you simply compare if holding the CCI30 was more profitable than your strategy; this is usually done when doing algorithms on quantopian as you choose your benchmark to compare; 

Basically if you trade eth/btc and can't beat the cci30 means you wasted your time (theoretically)

**Comment by xmatthias** [comment]
Same answer than in #1908 (it's not specific to this indicator / metric)

The problem i have with adding new columns to the backtest result is that we already have a 2 tables with 9 columns - which becomes quite difficult to interpret if you run it for more than a couple of pairs.

I personally think that a collection of code snippets everyone can run easily against their backtest results would be simpler to interpret and produce a less cluttered output (not everyone will be interrested in the same metrics, so everyone can run the part of the snippet they want to know).
My idea was to extend [this](https://www.freqtrade.io/en/latest/backtesting/#further-backtest-result-analysis) section in the documentation, and if it grows too big to move it to it's own page, or alternatively provide sample-notebooks (yes, jupyter, i know...) which users can easily run against their strategies.

**Comment by Dante2333** [comment]
i think this is some kind of basic information, anyone dealing with algorithmic trading has to check these variables; instead of adding it as a column, why not add it as a small square like the "sell count, ROI x times etc"

**Comment by xmatthias** [comment]
You say basic information ... you use cci30, others may use HODL BTC or sharp ratio or another indicator and don't care at all about this particular indicator. Not everyone uses the same thing for evaluation, which is good as otherwise everyone would use the same strategy.

if it's enough to have it for the summary (and not per pair) - then yes, that would be an option (also with several of these options).

**Comment by Dante2333** [comment]
indeed, adding holding BTC, holding the CCI30 are both really good benchmarks.

**Comment by hroff-1902** [comment]
It's a good enhancement in common, but @xmatthias is also right -- we cannot overload backtesting output with many raws...

**Comment by hroff-1902** [comment]
@xmatthias 

We could calculate these values, but do not print them if not requested. And add an option for additional columns, like `--columns="cci30, sharpe, whatelse-ratio, max-drawdown"`...

Another option is to add an interface for custom ratios and add as a sample a couple implementations of them...

**Comment by xmatthias** [comment]
an additional table (maybe before/after the sell-reason table) seems reasonable eventually.

I would limit it to ratios we can calculate from the backtest result / data and don't require additional data. cci30 and HODL would require data which may not  be available from some exchanges (not all offer <stake>/USD(T))  - so it'll be only sharpe, max-drawdown and "some others".

**Comment by Dante2333** [comment]
can't we simply download the CCI30 through their api @xmatthias ?

**Comment by xmatthias** [comment]
define "their API"?

**Comment by Dante2333** [comment]
@xmatthias sorry, they dont have an api, https://cci30.com/ajax/getIndexHistory.php <- they do provide this

**Comment by xmatthias** [comment]
the data is daily, which makes it non-comparable in my opinion.

Most users i know use rather short timeframes (between 5m-1h) - for which a comparison of daily cci30 values might be an estimation unless you select exact date-changes for your backtesting period.

I am also not sure on how you would compare this - use start-open and end-close and do the difference, comparing that with the backtest result?

However, feel free implement and do a PR on the calculation and we can discuss on how to include/display this properly.

**Comment by Dante2333** [comment]
Sorry, what does PR stands for?
I think you can still compare your strategy 5m 1h against cci30 unless you are running your backtest on specific hours?
Say you run your strategy on 20180101-20181231; it doesnt matter if its 5m or 1h, you compare it to inital buy of daily beginning (CCI30) against the end day of cci30 where you strategy also ends.

So if your strategy is cum profit is 140% but CCI30 cumuleative profit from (start-open) where yourperiod starts to end-close where the period for strategy ends; If (End-close/start-open)-1> 140% then it would have been better to buy the cci30 rather than use your strategy.

**Comment by xmatthias** [comment]
PR = Pull Request - basically that you do the change, we can then review and merge that chnage to the bot. 

yes(ish) - that's how i could imagine it too - however it'd be a "online" requirement for backtesting (query this file each time) - which i don't like as it'll slow backtesting down.

**Comment by hroff-1902** [comment]
found a way (a third-party site) how to fetch cci30 vals: http://www.cryptoindices.io/Pages/api/16

**Comment by hroff-1902** [comment]
Lykke's index -- composite of 25 cryptos, can be used as a benchmark index.
https://info.lykke.com

Lykke exchange available to ccxt.
```
$ python3 ./scripts/get_market_pairs.py lykke | grep LY
...
BTCLKK          BTC/LYKKE       BTC             LYKKE          
LYCIUSD         LYCI/USD        LYCI            USD            
LKKCHF          LYKKE/CHF       LYKKE           CHF            
LKKEUR          LYKKE/EUR       LYKKE           EUR            
LKKGBP          LYKKE/GBP       LYKKE           GBP            
LKKJPY          LYKKE/JPY       LYKKE           JPY            
LKKUSD          LYKKE/USD       LYKKE           USD            
PLYCIUSD        PLYCI/USD       PLYCI           USD            
SLYCIUSD        SLYCI/USD       SLYCI           USD            
```

**Comment by hroff-1902** [comment]
The list of indices (taken from https://hackernoon.com/the-ultimate-guide-to-cryptocurrency-indices-17ccf8fb0092):

* The Coinbase Index
* Bitwise10
* HODL10
* Crypto20
* HODL20
* CCI30
* HODL30
* CRIX
* The Bloomberg Galaxy Crypto Index

None of them are fetchable via an API, though...

**Comment by hroff-1902** [comment]
Maybe some crypto indices may be present at one of the sources, available for pandas_datareader: https://pandas-datareader.readthedocs.io/en/latest/remote_data.html

Available sources:

- Google Finance
- Morningstar
- IEX
- Robinhood
- Enigma
- Quandl
- St.Louis FED (FRED)
- Kenneth French’s data library
- World Bank
- OECD
- Eurostat
- Thrift Savings Plan
- Nasdaq Trader symbol definitions
- Stooq
- MOEX

These are most about stocks/forex, but cryptos are also available in some, in Yahoo Finance, for example...

**Comment by xmatthias** [comment]
I've removed the "Buy and Hold" comparison from the title of this issue.
We'll keep track of that part in #2524 - so the 2 can be worked on independently.

**Comment by xmatthias** [comment]
We've had this open now for 6 years - with nobody ever asking for it.

Closing this as i don't think it's of broader interest.
